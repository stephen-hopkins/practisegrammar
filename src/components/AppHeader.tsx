import Image from "next/image";
import Link from "next/link";
import logo from "public/pg-logo.png";

export default function AppHeader() {
  return (
    <header className="flex flex-row h-5rem justify-content-between">
      <Image src={logo} alt="application logo" className="img-logo h-5rem w-5rem" />
      <div className="bg-white w-5 mb-2 nav-background">
        <Link href="/about">About Us</Link>
      </div>
    </header>
  );
}
