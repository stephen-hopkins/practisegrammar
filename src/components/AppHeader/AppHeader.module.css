.header_wrapper {
  display: flex;
  flex-direction: row;
  height: 5rem;
  justify-content: space-between;
}

.img_logo {
  height: 4rem;
  width: 4rem;
  margin-top: 0.5rem;
  margin-left: 0.5rem;
  object-fit: contain;
}

.nav_background {
  position: relative;
  border-radius: 0 0 0 25px;
  width: fit-content;
  margin-bottom: 0.5rem;
  background-color: #ffffff;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.nav_background::before {
  content: "";
  position: absolute;
  background-color: transparent;
  left: -50px;
  height: 100px;
  width: 50px;
  border-top-right-radius: 50px;
  box-shadow: 0 -50px 0 0 white;
}

.nav_link {
  margin: 1rem;
}
