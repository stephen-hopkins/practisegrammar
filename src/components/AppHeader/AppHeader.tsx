"use client";

import Image from "next/image";
import Link from "next/link";
import logo from "public/pg-logo.png";
import styles from "./AppHeader.module.css";

export default function AppHeader() {
  return (
    <header className={styles.header_wrapper}>
      <Link href="/">
        <Image src={logo} alt="application logo" className={styles.img_logo} />
      </Link>
      <div className={styles.nav_background}>
        <Link className={styles.nav_link} href="/about">
          About Us
        </Link>
      </div>
    </header>
  );
}
