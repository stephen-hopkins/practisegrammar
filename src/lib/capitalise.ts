export function capitalise(word: string) {
  return word.charAt(0).toLocaleLowerCase() + word.slice(1);
}