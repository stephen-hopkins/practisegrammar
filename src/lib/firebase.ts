import { initializeApp } from "firebase/app";
import { Concept, Language } from "@/src/models/constants";
import { addDoc, collection, CollectionReference, getDocs, getFirestore, query, where } from "@firebase/firestore";
import { Sentence, SentenceInfo } from "@/src/models/translation";

const firebaseConfig = {
  apiKey: "AIzaSyCXVeg1hzBtWVW2FK9_cz7r7gygHIH8E2k",
  authDomain: "practisegrammar.firebaseapp.com",
  projectId: "practisegrammar",
  storageBucket: "practisegrammar.appspot.com",
  messagingSenderId: "666711626320",
  appId: "1:666711626320:web:410a88534639301d4b7755",
  measurementId: "G-YCLPJ0Z093",
};

function getDb() {
  // const analytics = getAnalytics(app);
  const app = initializeApp(firebaseConfig);
  const db = getFirestore(app);

  return {
    sentences: collection(db, "sentences") as CollectionReference<SentenceInfo>,
  };
}

export async function getSentence(language: Language, concept: Concept, word: string) {
  const { sentences } = getDb();
  const q = query(
    sentences,
    where("language", "==", language),
    where("concept", "==", concept),
    where("word", "==", word)
  );
  const querySnapshot = await getDocs(q);
  if (!querySnapshot.empty) {
    return querySnapshot.docs[0].data();
  }
}

export async function addSentence(sentence: Sentence) {
  const { translations, ...metaData } = sentence;
  const { sentences } = getDb();
  //const doc = await addDoc(sentences, metaData, "translations", translations);
  //await addDoc(collection(sentences, doc.id, "translations"), translations);
}
