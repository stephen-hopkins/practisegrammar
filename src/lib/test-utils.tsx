import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { RenderOptions, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { MemoryRouterProvider } from "next-router-mock/MemoryRouterProvider";

jest.mock("next/navigation", () => require("next-router-mock"));

const queryClient = new QueryClient();
const AllProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <MemoryRouterProvider url="/">{children}</MemoryRouterProvider>
    </QueryClientProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
