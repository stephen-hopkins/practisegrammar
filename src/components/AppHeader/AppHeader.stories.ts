import { Meta, StoryObj } from "@storybook/react";
import AppHeader from "./AppHeader";

const meta: Meta<typeof AppHeader> = {
  title: "AppHeader",
  component: AppHeader,
};

export default meta;

type Story = StoryObj<typeof AppHeader>;

export const Default: Story = {
  args: {},
};
