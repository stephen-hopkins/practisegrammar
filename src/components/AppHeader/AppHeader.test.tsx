import { render } from "@/src/lib/test-utils";
import AppHeader from "./AppHeader";
import userEvent from "@testing-library/user-event";
import mockRouter from "next-router-mock";

describe("AppHeader", () => {
  it("renders", () => {
    render(<AppHeader />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<AppHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

  // skipped for now because next-router-mock doesn't support next/navigation properly yet
  // https://github.com/scottrippey/next-router-mock/issues/67
  it.skip("clicking on logo takes you home", async () => {
    const user = userEvent.setup();
    mockRouter.push("/about");
    const { getByAltText } = render(<AppHeader />);

    const logo = getByAltText("application logo");
    await user.click(logo);

    expect(mockRouter.asPath).toBe("/");
  });
});
