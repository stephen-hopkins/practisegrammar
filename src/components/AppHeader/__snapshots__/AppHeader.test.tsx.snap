// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header_wrapper"
  >
    <a
      href="/"
    >
      <img
        alt="application logo"
        class="img_logo"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </a>
    <div
      class="nav_background"
    >
      <a
        class="nav_link"
        href="/about"
      >
        About Us
      </a>
    </div>
  </header>
</DocumentFragment>
`;
